src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -1499,17 +1564,31 @@ public <C extends GitCommand<?>> C authenticate(TransportCommand<C, ?> command)
     public List<CommitInfoDTO> getCommitInfos(VcsRepositoryUri vcsRepositoryUri) throws GitAPIException {
         List<CommitInfoDTO> commitInfos = new ArrayList<>();
 
-        try (var repo = getOrCheckoutRepository(vcsRepositoryUri, true); var git = new Git(repo)) {
-            var commits = git.log().call();
-            commits.forEach(commit -> {
-                var commitInfo = CommitInfoDTO.of(commit);
-                commitInfos.add(commitInfo);
-            });
+        if (profileService.isLocalVcsActive()) {
+            log.info("Using local VCS for getting commit info on repo {}", vcsRepositoryUri);
+            try (var repo = getBareRepository(vcsRepositoryUri); var git = new Git(repo)) {
+                getCommitInfo(git, commitInfos);
+            }
         }
+        else {
+            try (var repo = getOrCheckoutRepository(vcsRepositoryUri, true); var git = new Git(repo)) {
```

Comment:
Why do we log here before `getBareRepository` in this place but add the log for the else branch inside the method `getOrCheckoutRepository`?